index.js

```diff
@@ -11,6 +11,19 @@ const changePrototype = (to, from) => {
 	Object.setPrototypeOf(to, fromPrototype);
 };
 
+const wrappedToString = (withName, fromBody) => `/* Wrapped ${withName}*/\n${fromBody}`;
+
+const changeToString = (to, from, name) => {
+	const withName = name === '' ? '' : `with ${name.trim()}() `;
+	const value = wrappedToString.bind(null, withName, from.toString());
```

Comment:
We:
  - call `from.toString()` early (during `mimicFn()`) as opposed to lazily (during `to.toString()`). 
  - use `bind()` instead of a closure. 

This is to ensure `from` can be garbage collected. Otherwise `to.toString()` closure would contain a reference to `from`.

Also calling `from.toString()` early allows caching it in case `to.toString()` is called several times. I'm not sure whether that's already done by V8 under the hood though.